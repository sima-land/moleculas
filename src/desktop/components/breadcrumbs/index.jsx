import React, { useState, useRef, useEffect } from 'react';
import CSSTransition from 'react-transition-group/CSSTransition';
import Popup from '@dev-dep/ui-nucleons/popups/popup';
import on from '@dev-dep/ui-nucleons/helpers/on';
import classnames from 'classnames/bind';
import prop from 'lodash/fp/prop';
import DownSVG from './square-arrow-down.svg';
import styles from './breadcrumbs.scss';

const cx = classnames.bind(styles);

const DEFAULTS = {
  getName: prop('name'),
  getItemURL: prop('url'),
  getItemSiblings: prop('items'),
  isActiveItem: prop('isActive'),
  hasSiblings: breadcrumb => Array.isArray(breadcrumb?.items) && breadcrumb.items.length > 0,
};

/**
 * Список "хлебных крошек".
 * @param {Object} props Свойства компонента.
 * @param {Array} props.items Список "Хлебных крошек".
 * @param {Function} [props.getItemName] Вернёт имя элемента списка (по умолчанию свойство "name").
 * @param {Function} [props.getItemURL] Вернёт URL элемента списка (по умолчанию свойство "url").
 * @param {Function} [props.getItemSiblings] Вернёт список дочерних ссылок (по умолчанию свойство "items").
 * @param {Function} [props.isItemActive] Вернёт true если элемент представляет активный раздел.
 * @param {Function} [props.hasItemSiblings] Вернёт true если имеются соседние категории.
 * @param {Function} [props.onItemSiblingsShow] Будет вызвана при открытии списка соседних элементов.
 * @param {string} [props.'data-testid'] Идентификатор для систем автоматизированного тестирования.
 * @return {ReactElement} Элемент.
 */
export const Breadcrumbs = ({
  items,
  getItemName = DEFAULTS.getName,
  getItemURL = DEFAULTS.getItemURL,
  getItemSiblings = DEFAULTS.getItemSiblings,
  isItemActive = DEFAULTS.isActiveItem,
  hasItemSiblings = DEFAULTS.hasSiblings,
  onItemSiblingsShow,
  'data-testid': dataTestId,
}) => (
  <ul className={cx('breadcrumbs')} data-testid={dataTestId}>
    {Array.isArray(items) && items.map((breadcrumb, index) => (
      <Breadcrumb
        key={index}
        title={getItemName(breadcrumb)}
        href={getItemURL(breadcrumb)}
        canOpen={hasItemSiblings(breadcrumb)}
        siblings={getItemSiblings(breadcrumb)}
        active={isItemActive(breadcrumb)}
        getSiblingName={getItemName}
        isActiveSibling={isItemActive}
        getSiblingHref={getItemURL}
        onSiblingsShow={() => onItemSiblingsShow?.(breadcrumb)}
        data-testid={`breadcrumb:${index}`}
      />
    ))}
  </ul>
);

/**
 * Элемент списка "хлебных крошек".
 * @param {Object} props Свойства.
 * @return {ReactElement} Элемент.
 */
const Breadcrumb = ({
  title,
  href,
  siblings,
  active,
  getSiblingName,
  isActiveSibling,
  getSiblingHref,
  onSiblingsShow,
  canOpen,
  'data-testid': dataTestId,
}) => {
  const popupRef = useRef();
  const openerRef = useRef();
  const hasSiblings = siblings?.length > 0;
  const breadcrumbName = active
    ? (
      <span className={cx('active')}>{title}</span>
    ) : (
      <a
        className={cx('link')}
        href={href}
        children={title}
      />
    );

  const [withPopup, togglePopup] = useSiblingsPopup(popupRef, openerRef);

  return (
    <li data-testid={dataTestId}>
      {breadcrumbName}
      {canOpen && (
        <span ref={openerRef}>
          <DownSVG
            data-testid='breadcrumb:siblings-opener'
            className={cx('toggle-icon')}
            onClick={() => {
              togglePopup(true);
              onSiblingsShow?.();
            }}
          />
        </span>
      )}
      {hasSiblings && (
        <CSSTransition
          in={withPopup}
          timeout={200}
          unmountOnExit
          classNames={{
            enter: cx('fade-enter'),
            enterActive: cx('fade-enter-active'),
            exit: cx('fade-exit'),
            exitActive: cx('fade-exit-active'),
          }}
        >
          <Popup
            additionalClass={cx('popup')}
            ref={popupRef}
          >
            {breadcrumbName}
            <DownSVG
              data-testid='breadcrumb:siblings-closer'
              className={cx('toggle-icon', 'close')}
              onClick={() => togglePopup(false)}
            />
            <ul className={cx('siblings-list')}>
              {siblings.map((siblingItem, siblingIndex) => {
                const isActive = isActiveSibling(siblingItem);
                const siblingName = getSiblingName(siblingItem);

                return (
                  <li
                    key={siblingIndex}
                    className={cx('sibling-item')}
                    data-testid={`breadcrumbs-item-sibling:${siblingIndex}`}
                  >
                    {isActive ? (
                      <span className={cx('disabled')}>{siblingName}</span>
                    ) : (
                      <a
                        className={cx('link')}
                        color='dark-gray'
                        href={getSiblingHref(siblingItem)}
                        children={siblingName}
                      />
                    )}
                  </li>
                );
              })}
            </ul>
          </Popup>
        </CSSTransition>
      )}
    </li>
  );
};

/**
 * Хук работы с попапом соседних элементов "крошки".
 * @param {Object} popupRef Ref попапа.
 * @param {Object} openerRef Ref открывающего элемента.
 * @return {Function} Функция-обработчик монтирования компонента.
 */
const useSiblingsPopup = (popupRef, openerRef) => {
  const [withPopup, togglePopup] = useState(false);

  useEffect(() => {
    const offClick = on(window, 'click', ({ target }) => {
      target

        // если клик произошел не по узлу иконки открытия и не внутри popup'а
        && [popupRef.current, openerRef.current].every(node => node && node !== target && !node.contains(target))
        && togglePopup(false);
    });

    const offKeyDown = on(window, 'keydown', ({ code, keyCode }) => {
      (code === 'Escape' || keyCode === 27) && togglePopup(false);
    });

    return () => {
      offClick();
      offKeyDown();
    };
  }, []);

  return [withPopup, togglePopup];
};
