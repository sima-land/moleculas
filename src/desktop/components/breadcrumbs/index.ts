export { type Breadcrumb, type BreadcrumbsProps, Breadcrumbs } from './breadcrumbs';
