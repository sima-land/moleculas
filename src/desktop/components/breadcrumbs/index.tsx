import React, { useEffect, useState, useRef } from 'react';
import CSSTransition from 'react-transition-group/CSSTransition';
import { Plate } from '@sima-land/ui-nucleons/plate';
import { Link } from '@sima-land/ui-nucleons/link';
import { times } from 'lodash';
import classnames from 'classnames/bind';
import styles from './breadcrumbs.module.scss';
import DownSVG from './square-arrow-down.svg';
import on from '@sima-land/ui-nucleons/helpers/on';

export interface Sibling {
  name: string;
  url?: string;
  [key: string]: any;
}

export interface Breadcrumb extends Sibling {
  siblings?: {
    state: 'initial' | 'fetching' | 'ready';
    data?: Sibling[];
  };
}

export interface BreadcrumbsProps {
  /** Список хлебных крошек. */
  items: Breadcrumb[];

  /** Получив хлебную крошку должна вернуть true если крошка выбрана. */
  isSiblingActive?: (sibling: Sibling) => boolean;

  /** Сработает при открытии попапа со смежными ссылками. */
  onSiblingsPopupOpen?: (item: Breadcrumb) => void;

  /** Идентификатор для систем автоматизированного тестирования. */
  'data-testid'?: string;
}

const cx = classnames.bind(styles);

/**
 * Список "хлебных крошек".
 * @param props Свойства.
 * @return Элемент.
 */
export const Breadcrumbs = ({
  items,
  onSiblingsPopupOpen,
  isSiblingActive,
  'data-testid': testId = 'breadcrumbs',
}: BreadcrumbsProps) => (
  <nav className={cx('root')} data-testid={testId}>
    {items.map((breadcrumb, index) => (
      <Breadcrumb
        key={index}
        data={breadcrumb}
        onSiblingsPopupOpen={() => onSiblingsPopupOpen?.(breadcrumb)}
        isSiblingActive={sibling => Boolean(isSiblingActive?.(sibling))}
      />
    ))}
  </nav>
);

/**
 * Элемент списка "хлебных крошек".
 * @param props Свойства.
 * @return Элемент.
 */
const Breadcrumb = ({
  data,
  onSiblingsPopupOpen,
  isSiblingActive,
}: {
  data: Breadcrumb;
  isSiblingActive: (sibling: Breadcrumb) => boolean;
  onSiblingsPopupOpen: () => void;
}) => {
  const popupRef = useRef<HTMLDivElement>(null);

  const [withPopup, togglePopup] = useState<boolean>(false);

  // с useOutsideClick не работает после обновления React до 17 версии, пока так
  useEffect(() => {
    if (withPopup) {
      return on(window, 'mousedown', togglePopup.bind(null, false));
    }
  }, [withPopup]);

  const selfName = (
    <Link color='gray87' href={data.url}>
      {data.name}
    </Link>
  );

  return (
    <div data-testid='breadcrumb' className={cx('breadcrumb')}>
      {selfName}

      {data.siblings && (
        <>
          <DownSVG
            data-testid='breadcrumb:siblings-opener'
            className={cx('toggle-icon')}
            onClick={() => {
              togglePopup(true);
              onSiblingsPopupOpen();
            }}
          />

          <CSSTransition
            in={withPopup}
            timeout={200}
            unmountOnExit
            classNames={{
              enter: cx('fade-enter'),
              enterActive: cx('fade-enter-active'),
              exit: cx('fade-exit'),
              exitActive: cx('fade-exit-active'),
            }}
          >
            <Plate ref={popupRef} shadow='z3' rounds='m' className={cx('popup')}>
              <div className={cx('popup-header')}>
                {selfName}
                <DownSVG
                  data-testid='breadcrumb:siblings-closer'
                  className={cx('toggle-icon', 'close')}
                  onClick={() => togglePopup(false)}
                />
              </div>

              <ul className={cx('siblings')}>
                {data.siblings.state === 'fetching' && (
                  <>
                    {times(8).map(index => (
                      <div key={index} className={cx('sibling-placeholder')} />
                    ))}
                  </>
                )}

                {data.siblings.state === 'ready' && (
                  <>
                    {data.siblings.data?.map((sibling, index) => (
                      <li key={index} className={cx('sibling')} data-testid='breadcrumb-sibling'>
                        {isSiblingActive(sibling) ? (
                          <span className={cx('active')}>{sibling.name}</span>
                        ) : (
                          <Link
                            color='gray87'
                            className={cx('link')}
                            href={sibling.url}
                            children={sibling.name}
                          />
                        )}
                      </li>
                    ))}
                  </>
                )}
              </ul>
            </Plate>
          </CSSTransition>
        </>
      )}
    </div>
  );
};
