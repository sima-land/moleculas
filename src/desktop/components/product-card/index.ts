export { ProductCardProps } from './types';
export { ProductCard, HoverCard, reduceBaseInfo, reduceHoverInfo } from './product-card';
