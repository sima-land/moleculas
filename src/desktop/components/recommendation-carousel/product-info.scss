.row + .row {
  margin-top: 8px;
}

.image-row {
  position: relative;
  display: block;
  width: 100%;
  .image {
    width: 100%;
  }
  .quick-view-btn {
    position: absolute;
    right: 8px;
    top: 8px;
  }
}

.badges-row {
  display: flex;
  flex-wrap: wrap;
  max-height: 56px;
  overflow: hidden;
  margin-bottom: -8px;
  .badge {
    margin: 0 4px 8px 0;
  }
}

.prices-row {
  display: flex;
  align-items: center;
  height: 24px;
  .price {
    font-size: 16px;
    line-height: 24px;
    font-weight: 700;
  }
  .old-price {
    font-size: 12px;
    line-height: 16px;
    margin-left: 8px;
    font-weight: 400;
  }
}

.link {
  overflow: hidden;
  word-break: break-word;
  max-width: 100%;
  font-size: 14px;
  line-height: 20px;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
}
