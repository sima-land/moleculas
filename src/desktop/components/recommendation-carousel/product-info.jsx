import React from 'react';
import { Link } from '@dev-dep/ui-nucleons/link';
import { Price } from '@dev-dep/ui-nucleons/price';
import { Image } from '../../../common/components/image';
import { isNonEmptyArray } from '../../../common/helpers/is-non-empty';
import { Badge } from '../../../common/components/badge';
import classnames from 'classnames/bind';
import styles from './product-info.scss';
import { WishButton } from '../../../common/components/wish-button';
import { QuickViewButton } from '../../../common/components/quick-view-button';
import withPrevent from '@dev-dep/ui-nucleons/helpers/with-prevent';

const cx = classnames.bind(styles);

/**
 * Компонент базовой информации о товаре.
 * @param {Object} props Свойства компонента.
 * @return {ReactElement} Элемент.
 */
export const ProductInfo = ({
  badges,
  currencyGrapheme,
  favorite,
  imageSrc,
  name,
  oldPrice,
  onFavoriteClick,
  onQuickViewClick,
  price,
  url,
  withImageButtons,
}) => (
  <>
    <a className={cx('row', 'image-row')} href={url}>
      <Image
        className={cx('image')}
        alt={name}
        src={imageSrc}
      />
      {withImageButtons && (
        <>
          <WishButton
            checked={favorite}
            className={cx('on-image-btn', 'wish')}
            onClick={withPrevent(onFavoriteClick)}
          />
          <QuickViewButton
            className={cx('on-image-btn', 'quick-view')}
            onClick={withPrevent(onQuickViewClick)}
          />
        </>
      )}
    </a>

    {Boolean(price) && (
      <div className={cx('row', 'prices-row')}>
        <Price
          value={price}
          currencyGrapheme={currencyGrapheme}
          className={cx('price')}
        />
        {Boolean(oldPrice) && (
          <Price
            value={oldPrice}
            currencyGrapheme={currencyGrapheme}
            className={cx('old-price')}
            crossedOut
          />
        )}
      </div>
    )}

    {isNonEmptyArray(badges) && (
      <div className={cx('row', 'badges-row')}>
        {badges.map((badge, index) => (
          <Badge
            {...badge}
            key={index}
            className={cx('badge')}
          />
        ))}
      </div>
    )}

    <Link
      className={cx('row', 'link')}
      href={url}
      children={name}
      color='gray87'
    />
  </>
);
