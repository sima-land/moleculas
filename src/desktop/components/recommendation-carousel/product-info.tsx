import React from 'react';
import { Link } from '@dev-dep/ui-nucleons/link';
import { Price } from '@dev-dep/ui-nucleons/price';
import { Image } from '../../../common/components/image';
import { Badge } from '../../../common/components/badge';
import classnames from 'classnames/bind';
import styles from './product-info.scss';
import { QuickViewButton } from '../../../common/components/quick-view-button';
import withPrevent from '@dev-dep/ui-nucleons/helpers/with-prevent';
import { WithHint } from '@dev-dep/ui-nucleons/with-hint';
import { Product } from './types';

export interface ProductInfoProps extends Product {
  onQuickViewClick?: React.MouseEventHandler<SVGSVGElement>
  withQuickView?: boolean
}

const cx = classnames.bind(styles);

/**
 * Компонент базовой информации о товаре.
 * @param props Свойства компонента.
 * @return Элемент.
 */
export const ProductInfo = ({
  badges,
  currencyGrapheme,
  imageSrc,
  name,
  oldPrice,
  onQuickViewClick,
  price,
  url,
  withQuickView,
}: ProductInfoProps) => (
  <>
    <a className={cx('row', 'image-row')} href={url}>
      <Image
        className={cx('image')}
        alt={name}
        src={imageSrc}
      />
      {withQuickView && (
        <WithHint hint='Быстрый просмотр' direction='left'>
          {(ref, toggle) => (
            <QuickViewButton
              ref={ref as any}
              onMouseEnter={() => toggle(true)}
              onMouseLeave={() => toggle(false)}
              className={cx('quick-view-btn')}
              onClick={withPrevent(onQuickViewClick)}
            />
          )}
        </WithHint>
      )}
    </a>

    {Boolean(price) && (
      <div className={cx('row', 'prices-row')}>
        <Price
          value={price}
          currencyGrapheme={currencyGrapheme}
          className={cx('price')}
        />

        {oldPrice && (
          <Price
            value={oldPrice}
            currencyGrapheme={currencyGrapheme}
            className={cx('old-price')}
            crossedOut
          />
        )}
      </div>
    )}

    {Array.isArray(badges) && badges.length > 0 && (
      <div className={cx('row', 'badges-row')}>
        {badges.map((badge, index) => (
          <Badge
            {...badge}
            key={index}
            className={cx('badge')}
          />
        ))}
      </div>
    )}

    <Link
      className={cx('row', 'link')}
      href={url}
      children={name}
      color='gray87'
    />
  </>
);
