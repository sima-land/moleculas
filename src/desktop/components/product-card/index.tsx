import React, { Children, forwardRef, isValidElement, cloneElement, useState } from 'react';
import { ProductInfo, ProductInfoProps, Parts } from '../../../common/components/product-info';
import { Plate, PlateProps } from '@sima-land/ui-nucleons/plate';
import { useLayer } from '@sima-land/ui-nucleons/helpers/layer';
import cn from 'classnames';
import styles from './product-card.module.scss';

export type ProductCardChildren = React.ReactElement<ProductInfoProps, typeof ProductInfo>;

export interface ProductCardProps extends React.HTMLAttributes<HTMLDivElement> {
  children: ProductCardChildren;
}

export const HoverCard = forwardRef<HTMLDivElement | null, PlateProps>((props, ref) => {
  const layer = useLayer() + 1;

  return (
    <Plate
      {...props}
      style={{ ...props.style, zIndex: layer }}
      ref={ref}
      rounds='m'
      shadow='z4'
      className={cn(styles.card, props.className)}
      data-testid='product-card:hover-card'
    />
  );
});

export const ProductCard = ({ children, className, ...props }: ProductCardProps) => {
  const [hovered, toggle] = useState<boolean>(false);

  return (
    <div {...props} data-testid='product-card' className={cn(styles.root, className)}>
      <div data-testid='product-card:info' onMouseEnter={() => toggle(true)}>
        {reduceBaseInfo(children)}
      </div>

      {hovered && (
        <HoverCard onMouseLeave={() => toggle(false)}>{reduceHoverInfo(children)}</HoverCard>
      )}
    </div>
  );
};

/**
 * Фильтрует содержимое карточки для вывода состояния без наведения.
 * @param el Элемент.
 * @return Элемент.
 */
export const reduceBaseInfo = (el: ProductCardChildren) =>
  cloneElement(el, {
    children: Children.map(el.props.children, child => {
      let result;

      if (isValidElement(child)) {
        switch (child.type) {
          // игнорируем кнопки-иконки на изображении
          case Parts.Image:
            result = cloneElement<any>(child, { children: undefined });
            break;

          // игнорируем футер
          case Parts.Footer:
            result = null;
            break;

          // остальное оставляем без изменений
          default:
            result = child;
            break;
        }
      }

      return result;
    }),
  });

export const reduceHoverInfo = (el: ProductCardChildren) =>
  cloneElement(el, {
    // при наведении у изображения строго непрозрачность = 1
    children: Children.map(el.props.children, child =>
      isValidElement(child) && child.type === Parts.Image
        ? cloneElement(child, { opacity: 1 })
        : child,
    ),
  });
