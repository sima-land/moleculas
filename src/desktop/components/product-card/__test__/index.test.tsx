import React from 'react';
import { render } from '@testing-library/react';
import { Simulate } from 'react-dom/test-utils';
import { ProductCard } from '..';
import { ProductData } from '../../../../common/components/product-info';

describe('ProductCard', () => {
  const product: ProductData = {
    name: 'Lorem ipsum dolor sit amet consectetur, adipisicing elit. Debitis, nulla.',
    url: 'https://sima-land.ru',
    price: 1200,
    oldPrice: 1450,
    currencyGrapheme: '₽',
    imageSrc: 'https://picsum.photos/240',
    badges: [
      { color: '#b52ea8', fields: [{ type: 'text', value: '-56%' }] },
      { color: '#ff7200', fields: [{ type: 'text', value: '3 по цене 2' }] },
      { color: '#2962ff', fields: [{ type: 'text', value: 'Товар месяца' }] },
      { color: '#00b8d4', fields: [{ type: 'text', value: 'Акция' }] },
    ],
  };

  it('should renders correctly', () => {
    const { container } = render(
      <ProductCard
        style={{ width: '240px' }}
        data={product}
        className='test-class'
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('should handle slots in children', () => {
    const { container } = render(
      <ProductCard data={product}>
        <ProductCard.CartControl stepText='hello' markupText='world'>
          <button>Add to cart</button>
        </ProductCard.CartControl>
      </ProductCard>
    );

    expect(container).toMatchSnapshot();
  });

  it('should render cart control loading state', async () => {
    const { baseElement } = render(
      <ProductCard data={product}>
        <ProductCard.CartControl loading stepText='hello' markupText='world'>
          <button>Add to cart</button>
        </ProductCard.CartControl>
      </ProductCard>
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('should handle "onLinkClick" prop', async () => {
    const spy = jest.fn();

    const { findByTestId } = render(
      <ProductCard
        data={product}
        onLinkClick={spy}
      />
    );

    expect(spy).toBeCalledTimes(0);

    Simulate.click(await findByTestId('product-info:image-link'));

    expect(spy).toBeCalledTimes(1);

    Simulate.click(await findByTestId('product-info:name-link'));

    expect(spy).toBeCalledTimes(2);
  });

  it('should handle image button click callbacks', () => {
    const spy = jest.fn();

    const { container } = render(
      <ProductCard
        data={product}
        onQuickViewClick={spy}
      />
    );

    expect(spy).toBeCalledTimes(0);

    Simulate.click(container.querySelector('[data-testid="quick-view-button"]') as any);

    expect(spy).toBeCalledTimes(1);
  });

  it('should show/hide hint for quick view button', async () => {
    const { baseElement, findByTestId } = render(
      <ProductCard data={product} />
    );

    expect(baseElement.querySelectorAll('[data-testid="hint"]')).toHaveLength(0);

    Simulate.mouseEnter(await findByTestId('quick-view-button'));

    expect(baseElement.querySelectorAll('[data-testid="hint"]')).toHaveLength(1);

    Simulate.mouseLeave(await findByTestId('quick-view-button'));

    expect(baseElement.querySelectorAll('[data-testid="hint"]')).toHaveLength(0);
  });
});
