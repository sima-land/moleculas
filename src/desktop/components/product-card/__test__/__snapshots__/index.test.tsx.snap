// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard should render base info correctly 1`] = `
<div>
  <div
    class="root"
    data-testid="product-card"
  >
    <div
      data-testid="product-card:info"
    >
      <div
        class="root image-overlay"
      >
        <a
          class="image-link"
          data-testid="product-info:image-link"
          href="https://www.products.com/some-product"
        >
          <img
            class="image"
            data-testid="product-info:image"
            src="https://www.images.com/some-product/1"
          />
        </a>
      </div>
      <div
        class="prices height-limit"
        data-testid="product-info:prices"
      >
        <span
          class="root price"
          data-testid="product-info:price"
        >
          23 476,23 $
        </span>
        <span
          class="root old-price crossed-out"
          data-testid="product-info:old-price"
        >
          25 000 $
        </span>
      </div>
      <a
        class="link link line-clamp basic-gray87"
        data-testid="product-info:name-link"
        href="https://www.products.com/some-product"
      >
        Some product
      </a>
    </div>
  </div>
</div>
`;

exports[`ProductCard should render hover card correctly 1`] = `
<div>
  <div
    class="root"
    data-testid="product-card"
  >
    <div
      data-testid="product-card:info"
    >
      <div
        class="root image-overlay"
      >
        <a
          class="image-link"
          data-testid="product-info:image-link"
          href="https://www.products.com/some-product"
        >
          <img
            class="image"
            data-testid="product-info:image"
            src="https://www.images.com/some-product/1"
          />
        </a>
      </div>
      <div
        class="prices height-limit"
        data-testid="product-info:prices"
      >
        <span
          class="root price"
          data-testid="product-info:price"
        >
          23 476,23 $
        </span>
        <span
          class="root old-price crossed-out"
          data-testid="product-info:old-price"
        >
          25 000 $
        </span>
      </div>
      <a
        class="link link line-clamp basic-gray87"
        data-testid="product-info:name-link"
        href="https://www.products.com/some-product"
      >
        Some product
      </a>
    </div>
    <div
      class="root card shadow-z4 rounds-md__all"
      data-testid="product-card:hover-card"
      style="z-index: 1;"
    >
      <div
        class="root image-overlay"
      >
        <a
          class="image-link"
          data-testid="product-info:image-link"
          href="https://www.products.com/some-product"
        >
          <img
            class="image"
            data-testid="product-info:image"
            src="https://www.images.com/some-product/1"
            style="opacity: 1;"
          />
        </a>
      </div>
      <div
        class="prices height-limit"
        data-testid="product-info:prices"
      >
        <span
          class="root price"
          data-testid="product-info:price"
        >
          23 476,23 $
        </span>
        <span
          class="root old-price crossed-out"
          data-testid="product-info:old-price"
        >
          25 000 $
        </span>
      </div>
      <a
        class="link link line-clamp basic-gray87"
        data-testid="product-info:name-link"
        href="https://www.products.com/some-product"
      >
        Some product
      </a>
      <div
        class="footer"
        data-testid="product-info:footer"
      >
        <div
          class="control-wrapper"
        >
          <div
            class="control"
          >
            <button
              class="root size-m view-primary"
              data-testid="button"
            >
              В корзину
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
